pkg/corpusaccess/corpusaccessor.py

```diff
@@ -0,0 +1,27 @@
+import yaml
+
+# TODO: deal with changing corpus, ie. from uOttawa to Reuters 
+class CorpusAccessor:
+	def __init__(self, ctx):
+		self.ctx = ctx
+		self.documents = {}
+
+		with open(self.ctx.corpus_path, "r") as corpus_handle:
+			corpus_stream = yaml.load_all(corpus_handle, Loader=yaml.Loader)
+			for doc in corpus_stream:
+				self.documents[doc.id] = doc
```

Comment:
The thing I'm worried about with this is that we will be reinstantiating the object _for every new search query_. This probably isn't a problem with the UOttawa corpus, but will probably be painful with the larger Reuter's collection (even with UOttawa, you can still feel the lag from reading/unmarshaling the file).

Since the Corpus _will not_ change during the run of a program, ideally we should be able to load it into memory once, and keep it alive throughout the entire program run. To do that, we'll need to model this as a singleton class. Does that make sense? If you're really pressed for time I don't mind us doing it this way for now, with the understanding that we'll most likely have to come back later and optimize it for the larger collection

Reply:
Makes sense, I'll refactor it to be a singleton class!